go-controller/pkg/ovn/ovn.go

```diff
@@ -438,6 +440,16 @@ func (oc *Controller) iterateRetryPods() {
 		podTimer := podEntry.timeStamp.Add(time.Minute)
 		if now.After(podTimer) {
 			podDesc := fmt.Sprintf("[%s/%s/%s]", pod.UID, pod.Namespace, pod.Name)
+			// it could be that the Pod got deleted, but Pod's DeleteFunc has not been called yet, so don't retry
+			_, err := oc.watchFactory.GetPod(pod.Namespace, pod.Name)
```

Comment:
From the log you shared, it looked like the delete event was coming too late. This could be either:
1. the informer cache is behind, in which case this change doesn't help because you are checking the informer cache
2. the pod handler is stuck/bottlenecked, maybe due to heavy churn of relying on the single threaded namespace handler?

I'm wondering if you just pull in the patches for getting rid of waiting on namespaces, if your issue goes away. Could you give it a try please? By checking the informer cache here, we are basically saying we don't trust the way we distribute pod events to the handler, and I'm not sure its the best strategy long term.